samples/snippets/csharp/VS_Snippets_Remoting/Socket_Sync_Send_Receive/CS/source.cs

```diff
@@ -123,20 +123,31 @@ public static int SendReceiveTest3(Socket server)
     // using the overload that takes a buffer, offset, message size, and socket flags.
     public static int SendReceiveTest4(Socket server)
     {
-        byte[] msg = Encoding.UTF8.GetBytes("This is a test");
-        byte[] bytes = new byte[256];
         try
         {
+            byte[] msg = Encoding.UTF8.GetBytes("This is a test");
             // Blocks until send returns.
-            int byteCount = server.Send(msg, 0, msg.Length, SocketFlags.None);
-            Console.WriteLine("Sent {0} bytes.", byteCount);
+            int sentTotalLength = server.Send(msg, 0, msg.Length, SocketFlags.None);
+            Console.WriteLine("Sent {0} bytes.", sentTotalLength);
 
+            const int bufferLength = 256;
+            byte[] received = new byte[bufferLength];
+            int receivedTotalLength = 0;
             // Get reply from the server.
-            byteCount = server.Receive(bytes, 0, server.Available,
-                                       SocketFlags.None);
+            while (true)
+            {
+                int receivedLength = server.Receive(received, receivedTotalLength, bufferLength, SocketFlags.None);
+                if (receivedLength == 0)

```

Comment:
This will keep draining data and resizing the `received` array until the other endpoint closes it's socket, which is not clear from the code. This example now differs from the rest of the samples which intend to demonstrate a single call without deeper context on how do the peers cooperate.

Note that there is also `SendReceiveTest3`, which suffers from the same issue, but can't be refactored in the same way, considering the way it's being used in the `main` method.

Can't we instead change the sample in the following way?

```diff
- byteCount = server.Receive(bytes, 0, server.Available, SocketFlags.None);
+ byteCount = server.Receive(bytes, 0, bytes.Length, SocketFlags.None);
```

Reply:
> Can't we instead change the sample in the following way?

Yes, `bytes.Length` will work.

> This will keep draining data

The original code uses `server.Available`, which I guess intends to drain all data available from server. That's why I created the loop.